### PR TITLE
Add :beam-runners-direct-java:needsRunnerTests to javaPreCommit

### PR DESCRIPTION
**Pull request by kanterov**
Before, we didn't run tests annotated with `@Category(NeedsRunner.class)` as a part of "Java Pre Commit" Jenkins job.

Because we didn't run runner tests for a while, there are few of them failing. This PR includes attempts to fix some of them.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/) [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) </br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | --- | ---

### CODE REVIEW

**Comment by kennknowles** [comment]
For this PR, can I request deliberate separate commits for the independent fixes?

**Comment by kanterov** [comment]
@kennknowles absolutely, working on it

**Comment by kennknowles** [comment]
Or if no fixes are trivially forthcoming, one commit that adds `@Ignore` with links works fine. I just wanted to keep specific bits clear.

File: sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java [hunk 1]
```diff
@@ -641,7 +643,8 @@ private void testAdditionalOutput(IsBounded bounded) {
     p.run();
   }
 
-  @Test
+  @Ignore
```
Comment (kennknowles) [inline sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/SplittableDoFnTest.java]: Link to Jira here

**Comment by kanterov** [comment]
@kennknowles Please take a look. I did everything as one PR, but we can split it into separate ones.

**Comment by kennknowles** [comment]
Doing it as one PR is fine. This looks great.

**Pull request merged**